### ISSUE TITLE
Impossible routing out of business park near Sevenoaks, GB

### ISSUE THREAD

**Issue by martenjj** [osmandapp/OsmAnd#10027]
Start: North Downs Business Park, 51.31096 0.15477
End: M25 westbound 51.28615 0.11721
Routing profile: Internal, Car, prefer motorways, no vehicle parameters, no avoid roads

OsmAnd version: 3.8.3, but problem also happened with earlier versions
Device: Samsung S10, Android 10
Map: UK England 1 Sep 2020, also happened with earlier maps

See screen shot.  The only way out of the business park is along Lime Pit Lane.  However, the calculated route started off by going roughly northeastwards to A224 Polhill, which would mean driving up a 30 metre high cliff.  The purple recorded track shows the route that should be taken.

There are no restricted access roads between the starting point and Starhill Road.  There is a barrier=gate at OSM node https://www.openstreetmap.org/node/3325133722, but it has no access restrictions.  If the gate is causing the routing problem then maybe the best way to handle it would be to calculate the route as if access were allowed, but put up the "The destination is located in an area with private roads..." question for confirmation.

![osmand-sevenoaks-routing-issueXXXX](https://user-images.githubusercontent.com/10631760/96109867-0d5eb480-0ed7-11eb-90d2-dda0561c18d3.jpg)

**Comment by vshcherb** [comment]
Tricky navigation, probably from the point you start there is no correct car navigation at all

**Comment by aceman444** [comment]
Yes, he said there is a gate mapped. A gate by default is not passable. @martenjj you need to add access restrictions to the gate and also the roads behind it. If it is private road, then OsmAnd can't do much here (except the dialog you suggested), unless it adds some cleverness like "if starting point is inside a private road, find the shortest way out of the private area"...

**Comment by kevinp2** [comment]
I encountered this problem yesterday, when driving through remote Utah roads in the mountains without cell service.

There is a Winter Closure gate on the state highway which is sometimes closed because of heavy snowfall but is usually open:
https://www.openstreetmap.org/node/2876949305

This gate was missing the **access=yes** tag.  (I have since added it).

OSMAnd refused to route through the gate and wanted me to drive around it for 60 miles.  Fortunately, I had previously reviewed the highway, and so was able to drive "blind" through the road without getting lost.

I understand the assumption that a gate may be usually closed.  But perhaps OSMAnd could warn and offer the user the option to route through:

- Gates.
- Private roads.

... since these are often passable at least some of the time.
